src/auto-evo/selection_pressure/PredatorRoot.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace AutoEvo;
+
+using System;
+
+public class PredatorRoot : SelectionPressure
+{
+    // Needed for translation extraction
+    // ReSharper disable ArrangeObjectCreationWhenTypeEvident
+    public static readonly LocalizedString Name = new LocalizedString("PREDATOR_ROOT_PRESSURE");
+
+    // ReSharper restore ArrangeObjectCreationWhenTypeEvident
+
+    private static readonly Compound ATP = SimulationParameters.Instance.GetCompound("atp");
+    private static readonly Compound Glucose = SimulationParameters.Instance.GetCompound("glucose");
+
+    private Patch patch;
+
+    public PredatorRoot(Patch patch, float weight) : base(weight, [
+        AddOrganelleAnywhere.ThatConvertBetweenCompounds(Glucose, ATP),
+    ])
+    {
+        this.patch = patch;
+    }
+
+    public override float Score(Species species, SimulationCache cache)
+    {
+        if (species is not MicrobeSpecies microbeSpecies)
+            return 0;
+
+        var atpFromGlucose = cache.GetCompoundGeneratedFrom(Glucose, ATP, microbeSpecies, patch.Biome);
+
+        // Ensure that a predator actually needs the glucose from prey
+        if (atpFromGlucose >= cache.GetEnergyBalanceForSpecies(microbeSpecies, patch.Biome).TotalConsumption)
+        {
+            return 1;
+        }
+
+        if (atpFromGlucose >= cache.GetEnergyBalanceForSpecies(microbeSpecies, patch.Biome)
+                .TotalConsumptionStationary)
+        {
+            return 0.5f;
+        }
+
+        // for now we strictly forbid predators that need another food source to live
+        return 0;
+    }
+
+    public override float GetEnergy()
+    {
+        return 0;
+    }
+
+    public override IFormattable GetDescription()
+    {
+        // This shouldn't be called on 0 energy pressures
+        return Name;
+    }
+
+    public override string ToString()
+    {
+        return Name.ToString();
```

Comment:
Couldn't the base class have this as the default implementation as it seems to be the same in most of the pressure classes?

Reply:
Afaik, no. The way Name is defined prevents that.

Reply:
Can't name be an abstract property then to force it to exist and allow access to it in the base class?

Reply:
It could be made into a property, but it would be no longer static. I'm not sure if that is an issue.

Reply:
I guess one kind of hacky way would be to have the abstract property but then implement it as returning a private static field.

```c#
private static readonly LocalizedString name = ...;

public override LocalizedString Name => name;
```

Something along those lines. In some situations I haven't cared and just assigned the property individually for each instance but I guess in this case there would be a benefit to the above pattern to avoid a bunch of duplicate localized string objects in memory.